File: src/test/java/de/lukaseichler/recurrentneuralnetwork/activation/TanhActivationTest.java
package de.lukaseichler.recurrentneuralnetwork.activation;

import org.testng.annotations.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class TanhActivationTest {

    @Test
    public void apply() throws Exception {
        assertThat(new TanhActivation().apply(3.0)).isEqualTo(Math.tanh(3.0));
    }
}

File: src/test/scala/de/lukaseichler/recurrentneuralnetwork/activation/TanhActivationTest.scala
package de.lukaseichler.recurrentneuralnetwork.activation

import org.scalatest.Matchers
import org.scalatest.testng.TestNGSuite
import org.testng.annotations.Test

import scala.math.tanh

/**
 * @author Lukas Eichler
 */
class TanhActivationTest extends TestNGSuite with Matchers {

  @Test
  def apply() {
    new TanhActivation().apply(3.0) should be(tanh(3.0))
  }
}
